File: pset6/readability/readability.py
from cs50 import get_string


def main():
    # takes in text from the user
    text = get_string("Enter text: ")
    words = 1
    sentences = 0
    letters = 0
    
    # counts the letters, words and sentences
    for a in text:
        if a == " ":
            words += 1
        if a == "." or a == "!" or a == "?":
            sentences += 1
        if (ord(a.lower()) >= 97) and (ord(a.lower()) <= 122):
            letters += 1
    
    # calcs the sentences and letters per 100
    sentences = (sentences / words) * 100
    letters = (letters / words) * 100
    index = round((0.0588 * letters) - (0.296 * sentences) - 15.8)
    
    # prints grade in correct format
    if index < 1:
        print("Before Grade 1")
    elif index > 16:
        print("Grade 16+")
    else:
        print(f"Grade {index}")
    
    
main()

File: pset2/readability/readability.c
#import <cs50.h>
#import <stdio.h>
#import <string.h>
#import <ctype.h>
#import <math.h>

double letters(string text, int x);
double sentences(string text, int x);

int main(void)
{
    //asks user to input their text.
    string text = get_string("Text: ");
    int words = 1; 
    
    //counts the number of words in the text
    for (int i = 0, n = strlen(text); i < n; i++)
    {
        if (text[i] == ' ')
        {
            words++;
        }
    }
    
    double l = letters(text, words);
    double s = sentences(text, words);
    
    //printf ("%f\n", (0.0588 * l) - (0.296 * s) - 15.8);
    int grade = round((0.0588 * l) - (0.296 * s) - 15.8);
    //printf("%f %f %i\n", l, s, words);
    
    //decides what to output depending on the grade
    if (grade < 1)
    {
        printf("Before Grade 1\n");
    }
    else if (grade >= 16)
    {
        printf("Grade 16+\n");
    }
    else 
    {
        printf("Grade %i\n", grade);
    }
}

// calculates the average number of letters for every 100 words of text.
double letters(string text, int x)
{
    float nletters = 0;
    for (int i = 0, n = strlen(text); i < n; i++)
    {
        text[i] = tolower(text[i]); //converts all to lower case so comparison is easier
        if (text[i] >= 'a' && text[i] <= 'z')
        {
            nletters++;
        }
    }
    return nletters / x * 100;
}

// calculates the average number of sentences for every 100 words of text.
double sentences(string text, int x)
{
    float nsentences = 0; 
    for (int i = 0, n = strlen(text); i < n; i++)
    {
        if (text[i] == '.' || text[i] == '!' || text[i] == '?')
        {
            nsentences++; 
        }
    }
    if (nsentences == 0)
    {
        nsentences = 1;
    }
    return nsentences / x * 100; 
}